File: bin/change-port.rb
require 'yaml'

# Initialization
port = ARGV[0]
target_file = "#{__dir__}/../docker-compose.yml"

# Saving YAML file data as a hash
data = YAML.load_file(target_file)

# Modifying the port number
data["services"]["web"]["ports"][0] = "#{port}:3000"

# Saving hash into YAML file
File.write(target_file, data.to_yaml)

File: bin/change-port.py
import yaml, sys

fileDir = "../docker-compose.yml"

with open(fileDir, "r") as f:
  yamlData = yaml.safe_load(f)
  
yamlData["services"]["web"]["ports"][0] = sys.argv[1] + ":3000"

with open(fileDir, "w") as f:
  yaml.dump(yamlData, f, sort_keys=False)